ports/stm32/system_stm32.c

```diff
@@ -179,15 +179,32 @@ MP_WEAK void SystemClock_Config(void) {
 
     #if defined(STM32F4) || defined(STM32F7) || defined(STM32H7)
 
-    /* Enable Power Control clock */
-    #if defined(STM32H7A3xxQ) || defined(STM32H7B3xxQ)
-    MODIFY_REG(PWR->CR3, PWR_SUPPLY_CONFIG_MASK, PWR_CR3_SMPSEN);
+    #if defined(STM32H7) && defined(SMPS)
+    // H7 MCUs with SMPS must provide a power supply configuration.
+    MODIFY_REG(PWR->CR3, PWR_SUPPLY_CONFIG_MASK, MICROPY_HW_PWR_SMPS_CONFIG);
     #elif defined(STM32H7)
+    // H7 MCUs without SMPS, lock the power supply configuration update.
     MODIFY_REG(PWR->CR3, PWR_CR3_SCUEN, 0);
-    #else
+    #elif defined(STM32F4) || defined(STM32F7)
```

Comment:
I think this should be left as an `#else` because it's intended to capture all remaining options.  Eg if new MCUs are added to this part of the code, they should use this block.